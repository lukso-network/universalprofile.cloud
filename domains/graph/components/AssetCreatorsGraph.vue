<script setup lang="ts">
type Props = {
  asset?: Asset
}

const props = defineProps<Props>()
const { formatMessage } = useIntl()
const isLoaded = computed(() => props.asset && !props.asset?.isMetadataLoading)
const creators = computed(() => {
  return props.asset?.tokenCreatorsData?.length
    ? props.asset?.tokenCreatorsData
    : props.asset?.ownerData
      ? [props.asset?.ownerData]
      : []
})
</script>

<template>
  <template v-if="isLoaded">
    <div v-if="!!creators?.length" class="animate-fade-in">
      <div class="heading-inter-14-bold pb-2">
        {{ formatMessage('asset_creators_title') }}
      </div>
      <div class="grid grid-cols-1 gap-2 sm:flex sm:flex-wrap">
        <AssetCreatorGraph
          v-for="(creator, index) in creators"
          :key="index"
          :creator="creator"
          :asset="asset"
        />
      </div>
    </div>
  </template>
  <AppPlaceholderSection v-else slot-class="flex gap-4">
    <AppPlaceholderLine class="h-[48px] w-[170px]" />
    <AppPlaceholderLine class="h-[48px] w-[140px]" />
  </AppPlaceholderSection>
</template>
