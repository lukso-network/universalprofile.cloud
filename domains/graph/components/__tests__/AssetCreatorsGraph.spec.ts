import { renderSuspended } from '@nuxt/test-utils/runtime'
import { beforeEach, describe, expect, it, vi } from 'vitest'

import AssetCreatorsGraph from '@/domains/graph/components/AssetCreatorsGraph.vue'

describe('AssetCreatorsGraph', () => {
  beforeEach(() => {
    useIntl().setupIntl(defaultConfig)
  })

  it('should render loading state', async () => {
    const component = await renderSuspended(AssetCreatorsGraph)

    expect(component.html()).toMatchSnapshot()
  })

  it('should render empty state', async () => {
    const component = await renderSuspended(AssetCreatorsGraph, {
      props: {
        asset: {
          tokenCreatorsData: [],
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render creators', async () => {
    const component = await renderSuspended(AssetCreatorsGraph, {
      props: {
        asset: {
          tokenCreatorsData: [
            {
              address: '0x123',
              name: 'John Doe',
              issuedAssets: ['0x456'],
            },
            {
              address: '0x789',
              name: 'Jane Doe',
              issuedAssets: ['0xabc'],
            },
          ],
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should fallback to owner', async () => {
    const component = await renderSuspended(AssetCreatorsGraph, {
      props: {
        asset: {
          tokenCreatorsData: [],
          ownerData: {
            address: '0x123',
            name: 'John Doe',
            issuedAssets: ['0x456'],
          },
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })
})
