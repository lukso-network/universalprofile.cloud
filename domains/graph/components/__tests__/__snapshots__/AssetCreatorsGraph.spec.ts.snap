// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetCreatorsGraph > should fallback to owner 1`] = `
"<div id="test-wrapper">
  <div class="animate-fade-in">
    <div class="heading-inter-14-bold pb-2">Creators</div>
    <div class="grid grid-cols-1 gap-2 sm:flex sm:flex-wrap">
      <lukso-button custom-class="px-4" variant="secondary">
        <div class="flex w-full items-center justify-between">
          <div class="flex items-center">
            <lukso-profile size="x-small" profile-address="0x123" has-identicon=""></lukso-profile>
            <lukso-username name="John Doe" address="0x123" size="small" class="pl-2" name-color="neutral-20"></lukso-username>
          </div>
          <div class="flex items-center">
            <lukso-tooltip variant="danger" text="Un-verified creator" class="ml-2">
              <lukso-icon name="cross-filled" color="red-55" secondary-color="neutral-100" size="small"></lukso-icon>
            </lukso-tooltip>
          </div>
        </div>
      </lukso-button>
    </div>
  </div>
</div>"
`;

exports[`AssetCreatorsGraph > should render creators 1`] = `
"<div id="test-wrapper">
  <div class="animate-fade-in">
    <div class="heading-inter-14-bold pb-2">Creators</div>
    <div class="grid grid-cols-1 gap-2 sm:flex sm:flex-wrap">
      <lukso-button custom-class="px-4" variant="secondary">
        <div class="flex w-full items-center justify-between">
          <div class="flex items-center">
            <lukso-profile size="x-small" profile-address="0x123" has-identicon=""></lukso-profile>
            <lukso-username name="John Doe" address="0x123" size="small" class="pl-2" name-color="neutral-20"></lukso-username>
          </div>
          <div class="flex items-center">
            <lukso-tooltip variant="danger" text="Un-verified creator" class="ml-2">
              <lukso-icon name="cross-filled" color="red-55" secondary-color="neutral-100" size="small"></lukso-icon>
            </lukso-tooltip>
          </div>
        </div>
      </lukso-button>
      <lukso-button custom-class="px-4" variant="secondary">
        <div class="flex w-full items-center justify-between">
          <div class="flex items-center">
            <lukso-profile size="x-small" profile-address="0x789" has-identicon=""></lukso-profile>
            <lukso-username name="Jane Doe" address="0x789" size="small" class="pl-2" name-color="neutral-20"></lukso-username>
          </div>
          <div class="flex items-center">
            <lukso-tooltip variant="danger" text="Un-verified creator" class="ml-2">
              <lukso-icon name="cross-filled" color="red-55" secondary-color="neutral-100" size="small"></lukso-icon>
            </lukso-tooltip>
          </div>
        </div>
      </lukso-button>
    </div>
  </div>
</div>"
`;

exports[`AssetCreatorsGraph > should render empty state 1`] = `
"<div id="test-wrapper">
  <!--v-if-->
</div>"
`;

exports[`AssetCreatorsGraph > should render loading state 1`] = `
"<div id="test-wrapper">
  <div>
    <div class="mb-3 h-[17px] w-1/2 animate-pulse rounded-4 bg-neutral-90"></div>
    <div class="flex gap-4">
      <div class="animate-pulse rounded-4 bg-neutral-90 h-[48px] w-[170px]"></div>
      <div class="animate-pulse rounded-4 bg-neutral-90 h-[48px] w-[140px]"></div>
    </div>
  </div>
</div>"
`;
